core/sequencer/server.go

```diff
@@ -68,12 +68,21 @@ func createMetrics(mf monitoring.MetricFactory) {
 		domainIDLabel)
 }
 
+// Queue reads messages that haven't been deleted off the queue.
+type Queue interface {
```

Comment:
Fair question. 
At the moment, there are separate interfaces for the `MutationQueue` and `MutationStorage`  because the existing / deprecated design puts things into a queue, deletes them from the queue and copies them to a separate mutation storage table, and the queue and storage tables can be / are implemented by separate packages. 

The new design I'm migrating to just writes queued items in time order, removing the need for a second mutations storage object, but that's a few PRs away. 

Also worth noting: I'm moving the interfaces to be closer to their consumers rather than their implementations.  https://blog.chewxy.com/2018/03/18/golang-interfaces/